### PR TITLE
monitoring: resolve metric labeling TODOs

### PR DESCRIPTION
**Pull request by uwedeportivo**
Resolves TODOs left over from https://github.com/sourcegraph/sourcegraph/pull/6586

Fixes https://github.com/sourcegraph/sourcegraph/issues/6618

### CODE REVIEW

**Comment by uwedeportivo** [comment]
https://github.com/sourcegraph/sourcegraph/issues/6599

File: internal/api/internal_client.go [hunk 1]
```diff
@@ -343,29 +343,30 @@ func (c *internalClient) meteredPost(ctx context.Context, route string, reqBody,
...
 // respBody is non-nil the response body will be JSON unmarshalled to resp.
-func (c *internalClient) post(ctx context.Context, route string, reqBody, respBody interface{}) error {
+func (c *internalClient) post(ctx context.Context, route string, reqBody, respBody interface{}) (int, error) {
 	var data []byte
 	if reqBody != nil {
 		var err error
 		data, err = json.Marshal(reqBody)
 		if err != nil {
-			return err
+			return 0, err
```
Comment (unknwon) [inline internal/api/internal_client.go]: nit: I think `-1` is better, which clearly indicates an impossible status code. Similar to `*Index` return `-1` for not found case. WDYT?

File: internal/api/internal_client.go [hunk 2]
```diff
@@ -343,29 +343,30 @@ func (c *internalClient) meteredPost(ctx context.Context, route string, reqBody,
...
 		}
 	}
 
 	resp, err := ctxhttp.Post(ctx, nil, c.URL+route, "application/json", bytes.NewBuffer(data))
 	if err != nil {
-		return err
+		return 0, err
```
Comment (unknwon) [inline internal/api/internal_client.go]: nit: same as above comment.

File: internal/api/internal_client.go [hunk 3]
```diff
@@ -343,29 +343,30 @@ func (c *internalClient) meteredPost(ctx context.Context, route string, reqBody,
...
 	defer resp.Body.Close()
 	if err := checkAPIResponse(resp); err != nil {
-		return err
+		return resp.StatusCode, err
 	}
 
 	if respBody != nil {
-		return json.NewDecoder(resp.Body).Decode(respBody)
+		err = json.NewDecoder(resp.Body).Decode(respBody)
+		return resp.StatusCode, err
```
Comment (unknwon) [inline internal/api/internal_client.go]:
```go
		return resp.StatusCode, json.NewDecoder(resp.Body).Decode(respBody)
```

Comment (uwedeportivo) [reply internal/api/internal_client.go]: you did that on the phone ?  :-)

Comment (unknwon) [reply internal/api/internal_client.go]: Apparently... not 😂

Comment (uwedeportivo) [reply internal/api/internal_client.go]: done

File: internal/api/internal_client.go (hunk 1 above)
Comment (uwedeportivo) [reply internal/api/internal_client.go]: i usually return the zero value of the type if it is supposed to be ignored. i don't like magic values. but zero means something here so i guess you're right. will do it

Comment (uwedeportivo) [reply internal/api/internal_client.go]: done

File: internal/api/internal_client.go (hunk 2 above)
Comment (uwedeportivo) [reply internal/api/internal_client.go]: done

**Comment by unknwon** [comment]
LGTM, assume it works :)

**Comment by uwedeportivo** [comment]
i tested by hand both phabricator and internal_client and made sure metrics show up with labels

**Pull request merged**